src/app/settings/views/Security/SessionTimeout/SessionTimeout.test.tsx

```diff
@@ -0,0 +1,133 @@
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import configureStore from "redux-mock-store";
+
+import SessionTimeout, {
+  Labels as SessionTimeoutLabels,
+} from "./SessionTimeout";
+
+import { actions as configActions } from "app/store/config";
+import type { RootState } from "app/store/root/types";
+import { renderWithBrowserRouter, getTestState } from "testing/utils";
+
+const mockStore = configureStore<RootState>();
+
+describe("SessionTimeout", () => {
+  let state: RootState;
+
+  beforeEach(() => {
+    state = getTestState();
+  });
+
+  it("displays a spinner while loading", () => {
+    state.config.loaded = false;
+    state.config.loading = true;
+    renderWithBrowserRouter(<SessionTimeout />, { state });
+
+    expect(screen.getByText(SessionTimeoutLabels.Loading)).toBeInTheDocument();
+  });
+
+  it("displays the form when the 'Edit' button is clicked", async () => {
+    renderWithBrowserRouter(<SessionTimeout />, { state });
+
+    expect(
+      screen.getByRole("form", {
+        name: SessionTimeoutLabels.ConfigureSessionTimeout,
+      })
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole("textbox", { name: SessionTimeoutLabels.Expiration })
+    ).toHaveValue("14 days");
+  });
+
+  it("displays the updated timeout length when the value is saved", async () => {
+    renderWithBrowserRouter(<SessionTimeout />, { state });
+
+    await userEvent.clear(
+      screen.getByRole("textbox", { name: SessionTimeoutLabels.Expiration })
+    );
+
+    await userEvent.type(
+      screen.getByRole("textbox", { name: SessionTimeoutLabels.Expiration }),
+      "3 hours"
+    );
+
+    await userEvent.click(
+      screen.getByRole("button", { name: SessionTimeoutLabels.Save })
+    );
+
+    expect(
+      screen.getByRole("textbox", { name: SessionTimeoutLabels.Expiration })
+    ).toHaveValue("3 hours");
```

Comment:
I don't think this does what you expect. The test isn't waiting for the result of the submission so this will always pass. You'd need to mock formik form saved state (e.g. with the `mockFormikFormSaved` util) in order to test this.

Reply:
Ahhh okay that makes sense. I'll rectify that